src/test/java/com/egov/business-services/tests/billing-service-demand.feature

```diff
@@ -41,125 +41,390 @@ Scenario: Test to Create Demand with valid field values
+    * match billingServiceDemandResponseBody.Demands != '#[0]'
+
+@search_NoBusinessService_05 @negative @billingServiceDemandSearch @billingServiceDemand
+Scenario: Test to Search Demand with no business service
+    * call read('../preTests/billingServiceDemandPretest.feature@successCreate')
+    * print billingServiceDemandResponseBody
+    * def demandId = billingServiceDemandResponseBody.Demands[0].id
+    * def searchDemandParams =
+    """
+        {
+            tenantId: '#(tenantId)',
+            consumerCode: '#(consumerCode)',
+            demandId: '#(demandId)'
+        }
+    """
+    * call read('../preTests/billingServiceDemandPretest.feature@searchDemandGenericParam')
+    * print billingServiceDemandResponseBody
+    * assert billingServiceDemandResponseBody.ResponseInfo.status == commonConstants.expectedStatus.ok
+    * assert searchDemandResponseStatus == 200
+    * match billingServiceDemandResponseBody.Demands != '#[0]'
+
+@search_InvalidTenantId_06 @negative @billingServiceDemandSearch @billingServiceDemand
+Scenario: Test to Search Demand with invalid tenant id
+    * call read('../preTests/billingServiceDemandPretest.feature@successCreate')
+    * print billingServiceDemandResponseBody
+    * def demandId = billingServiceDemandResponseBody.Demands[0].id
+    * def tenantId = commonConstants.invalidParameters.invalidTenantId
+    * call read('../preTests/billingServiceDemandPretest.feature@searchDemand')
+    * print billingServiceDemandResponseBody
+    * assert searchDemandResponseStatus == 403
+    * assert billingServiceDemandResponseBody.Errors[0].message == commonConstants.errorMessages.invalidTenantIdError
+    
+@search_NoTenantId_07 @negative @billingServiceDemandSearch @billingServiceDemand
+Scenario: Test to Search Demand with no tenant id
+    * call read('../preTests/billingServiceDemandPretest.feature@successCreate')
+    * print billingServiceDemandResponseBody
+    * def demandId = billingServiceDemandResponseBody.Demands[0].id
+    * def searchDemandParams =
+    """
+        {
+            businessService: '#(businessService)',
+            consumerCode: '#(consumerCode)',
+            demandId: '#(demandId)'
+        }
+    """
+    * call read('../preTests/billingServiceDemandPretest.feature@searchDemandGenericParam')
+    * print billingServiceDemandResponseBody
+    * assert searchDemandResponseStatus == 400
+    * assert billingServiceDemandResponseBody.Errors[0].message == commonConstants.errorMessages.nullParameterError
+
+@search_WithOnlyTenantId_08 @negative @billingServiceDemandSearch @billingServiceDemand
+Scenario: Test to Search Demand with only tenant id
+    * call read('../preTests/billingServiceDemandPretest.feature@successCreate')
+    * print billingServiceDemandResponseBody
+    * def demandId = billingServiceDemandResponseBody.Demands[0].id
+    * def searchDemandParams =
+    """
+        {
+            'tenantId': '#(tenantId)'
+        }
+    """
```

Comment:
A suggestion, instead of multiline we can also pass inline params like below. May be you can follow this later.
```
def searchDemandParams = {tenantId: '#(tenantId)'}
```